The GNU/Linux build system for scrotwm is, according to the author's own
words, "busted".

This replacement Makefile behaves in a sane way, and follows most of the
conventions established in the GNU world. It will be included upstream in
a future release.
Index: scrotwm/linux/Makefile
===================================================================
--- scrotwm.orig/linux/Makefile	2009-09-20 16:24:44.000000000 +0200
+++ scrotwm/linux/Makefile	2009-09-20 16:44:56.000000000 +0200
@@ -1,25 +1,32 @@
-# $scrotwm: Makefile,v 1.2 2009/02/08 04:10:00 marco Exp $ 
+PREFIX= /usr/local
+BINDIR= $(PREFIX)/bin
+MANDIR= $(PREFIX)/share/man/man1
+LIBDIR= $(PREFIX)/lib/scrotwm
 
-CFLAGS+= -Wall -ggdb -D_GNU_SOURCE -I.
+CFLAGS+= -Wall -ggdb -D_GNU_SOURCE -I. \
+		 -DSWM_LIB=\"$(DESTDIR)$(LIBDIR)/swmhack.so\"
 LDADD+= -lX11 -lXrandr
 
 CC= gcc
 
-LVERS= $(shell . ../lib/shlib_version; echo $$major.$$minor)
+all: scrotwm swmhack.so
 
-all: symlink scrotwm libswmhack
-
-symlink:
-	ln -sf ../scrotwm.c ../lib/swm_hack.c .
-
-scrotwm: scrotwm.o linux.o
+scrotwm: ../scrotwm.o linux.o
 	$(CC) $(LDFLAGS) -o $@ $+ $(LDADD)
 
 %.so: %.c
-	$(CC) $(CFLAGS) -c -fpic -DPIC $+ -o $@
+	$(CC) $(CFLAGS) -c -fpic -DPIC -o $@ $+
 
-libswmhack: swm_hack.so
-	$(CC) -shared -fpic -o libswmhack.so.$(LVERS) `lorder swm_hack.so|tsort`
+swmhack.so: ../lib/swm_hack.so
+	$(CC) -shared -fpic -o $@ $+
+
+install: all
+	install -m 0755 -D scrotwm $(DESTDIR)$(BINDIR)/scrotwm
+	install -m 0644 -D ../scrotwm.1 $(DESTDIR)$(MANDIR)/scrotwm.1
+	install -m 0644 -D ../scrotwm.conf $(DESTDIR)/etc/scrotwm.conf
+	install -m 0644 -D swmhack.so $(DESTDIR)$(LIBDIR)/swmhack.so
 
 clean:
-	rm -f scrotwm *.o *.so libswmhack.so.*
+	rm -f scrotwm swmhack.so *.o *.so ../*.o ../lib/*.so
+
+.PHONY: all install clean
